app/api/api_v1/routers/search.py

```diff
@@ -156,15 +157,20 @@ def search_documents(
         },
     )
 
-    if not validate_corpora_ids(db, allowed_corpora_ids):
-        msg = "Error validating corpora IDs."
+    # First corpora validation is app token against DB. At least one of the app token
+    # corpora IDs must be present in the DB to continue the search request.
+    if not verify_any_corpora_ids_in_db(db, allowed_corpora_ids):
+        msg = "Error verifying corpora IDs."
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail=msg,
             headers={"WWW-Authenticate": "Bearer"},
         )
 
-    _LOGGER.info("Starting search...")
+    _LOGGER.info(
```

Comment:
Is it me ... I can't see the second stage of the verification - validating the search requests' corpora ids against the allowed?

Reply:
Ah the PR title is wrong - the PR body says this is part one of 2 for validating corpora IDs.